app/services/mosaic_service.rb

```diff
@@ -10,13 +10,16 @@ def initialize(resource:)
     @resource = resource.decorate
   end
 
-  # calculate the path for the non-fingerprinted version of the file
   def path
     raise Error if raster_file_sets.empty?
-    mosaic_path = Valkyrie::Storage::Disk::BucketedStorage.new(base_path: base_path).generate(resource: resource, original_filename: "mosaic.json", file: nil).to_s
+    mosaic_path = Valkyrie::Storage::Disk::BucketedStorage.new(base_path: base_path).generate(resource: resource, original_filename: fingerprinted_filename, file: nil).to_s
     return mosaic_path if storage_adapter.find_by(id: mosaic_file_id)
   rescue Valkyrie::StorageAdapter::FileNotFound
-    build_node if MosaicGenerator.new(output_path: tmp_file.path, raster_paths: raster_paths).run
+    # build default mosaic file
+    build_node(default_filename) if MosaicGenerator.new(output_path: tmp_file.path, raster_paths: raster_paths).run
+
+    # build fingerprinted mosaic file
+    build_node(fingerprinted_filename) if MosaicGenerator.new(output_path: tmp_file.path, raster_paths: raster_paths).run
```

Comment:
Does this run the mosaic generation a second time? Can we use the created mosaic from the previous step?

Reply:
Good catch. We should be able to do that unless the temp file is deleted in between.